WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -225,19 +263,47 @@ public void SwapLabel(LabelViewModel labelViewModel)
 
 	private void OnSelectionChanged()
 	{
+		_hiddenIncludedPockets.Clear();
+
 		Money sumOfWhiteList =
 			NonPrivatePockets
 				.Where(pocket => pocket.Labels.All(pocketLabel => LabelsWhiteList.Any(labelViewModel => pocketLabel == labelViewModel.Value)))
 				.Sum(x => x.Amount);
 
-		_includePrivatePocket = NonPrivatePockets.Sum(x => x.Amount) < _targetAmount || (LabelsWhiteList.IsEmpty() && _privatePocket.Amount >= _targetAmount);
-		var totalSelected = sumOfWhiteList + (_includePrivatePocket ? _privatePocket.Amount : Money.Zero);
+		if (IsPrivatePocketNeeded())
+		{
+			_hiddenIncludedPockets.Add(_privatePocket);
+		}
+		else if (IsSemiPrivatePocketNeeded())
+		{
+			_hiddenIncludedPockets.Add(_semiPrivatePocket);
+		}
+		else if (IsPrivateAndSemiPrivatePocketNeeded())
+		{
+			_hiddenIncludedPockets.Add(_privatePocket);
+			_hiddenIncludedPockets.Add(_semiPrivatePocket);
+		}
+
+		var totalSelected = sumOfWhiteList + _hiddenIncludedPockets.Sum(x => x.Amount);
+
 		EnoughSelected = totalSelected >= _targetAmount;
 
 		this.RaisePropertyChanged(nameof(LabelsWhiteList));
 		this.RaisePropertyChanged(nameof(LabelsBlackList));
 	}
 
+	private bool IsPrivatePocketNeeded() =>
+		(NonPrivatePockets.Sum(x => x.Amount) < _targetAmount && _privatePocket.Amount + _semiPrivatePocket.Amount < _targetAmount && NonPrivatePockets.Sum(x => x.Amount) + _privatePocket.Amount >= _targetAmount) ||
+		(LabelsWhiteList.IsEmpty() && _privatePocket.Amount >= _targetAmount);
```

Comment:
These calculations also completely disregard fees.

Reply:
If an exception happens regarding this during transaction building, the `_targetAmount` gets adjusted by the missing amount and it will try to find pockets again. This happens in the background, the user doesn't notice anything. But, yes it can be improved https://github.com/zkSNACKs/WalletWasabi/pull/8344#discussion_r894997375.